docs/architecture/grpc-for-wcf-developers/self-hosted.md

```diff
@@ -1,11 +1,185 @@
+
+The service file uses INI format, as shown in this example.
+
+```ini
+[Unit]
+Description=My gRPC Application
+
+[Service]
+Type=notify
+ExecStart=/usr/sbin/myapp
+
+[Install]
+WantedBy=multi-user.target
+```
+
+The `Type=notify` property tells `systemd` that the application will notify it on startup and shutdown. The `WantedBy=multi-user.target` setting will cause the service to start when the Linux system reaches "runlevel 2", meaning a non-graphical multi-user shell is active.
+
+Before `systemd` will recognize the service, it needs to reload its configuration. You control `systemd` using the `systemctl` command. After reloading, use the `status` sub-command to confirm the application has registered successfully.
+
+```console
+sudo systemctl daemon-reload
+sudo systemctl status myapp
+```
+
+If you have configured the service correctly, the following output will be shown.
+
+```text
+myapp.service - My gRPC Application
+ Loaded: loaded (/etc/systemd/system/myapp.service; disabled; vendor preset: enabled)
+ Active: inactive (dead)
+```
+
+Use the `start` command to start the service.
+
+```console
+sudo systemctl start myapp.service
+```
+
+> [!TIP]
+> The `.service` extension is optional when using `systemctl start`.
+
+To tell `systemd` to start the service automatically on system startup, use the `enable` command.
+
+```console
+sudo systemctl enable myapp
+```
+
+### Logging to journald
+
+The Linux equivalent of the Windows Event Log is `journald`, a structured logging system service that is part of `systemd`. Log messages written to the standard output by a Linux daemon are automatically written to `journald`, so to configure logging levels etc. use the `Console` section of the logging configuration. The `UseSystemd` host builder method automatically configures the console output format to suit the journal.
+
+Because `journald` is the standard for Linux logs, there are a variety of tools that integrate with it, and you can easily route logs from `journald` to an external logging system. Working locally on the host, you can use the `journalctl` command to view logs from the command line.
+
+```console
+sudo journalctl -u myapp
+```
+
+> [!TIP]
+> If you have a GUI environment available on your host, a few graphical log viewers are available for Linux, such as *QJournalctl* and *gnome-logs*.
+
+To learn more about querying the systemd journal from the command line with journalctl refer to [the man pages](https://manpages.debian.org/buster/systemd/journalctl.1).
+
+## HTTPS Certificates for self-hosted applications
+
+When running a gRPC application in production you should use a proper SSL certificate from a trusted Certificate Authority (CA). This could be a public CA, or an internal one for your organization. To use the certificate with an ASP.NET Core 3.0 application, make sure it is in PKCS #12 format (a `.pfx` file), protected by a strong password.
```

Comment:
Certificates aren't my forte so I could be wrong, but is using a cert from the cert store recommended? // @shirhatti

Reply:
I think cert store is a Windows thing, maybe needs a note to say use store on Windows and pfx file on Linux?

Reply:
Maybe?! @shirhatti what is best practice?

Reply:
Definitely use the cert store on Windows! We support pfx files on Linux, but have always shied away from making production recommendations

Reply:
To be more accurate, we don't require a pfx file. We just require an X509Certificate and a pfx file is just one way to get to one- https://docs.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509certificate.-ctor?view=netcore-3.0